src/attribute.rs

```diff
@@ -654,6 +681,24 @@ impl<'a> Attribute<'a> {
                         x.as_ptr() as *const _,
                     )
                 }),
+                AttrValue::Strs(ref x) => {
+                    let mut cstrings: Vec<CString> = Vec::with_capacity(x.len());
```

Comment:
This is okay, but I think you can also use iterators here, see https://github.com/georust/gdal/pull/239/files#diff-c9786f4e33a40ebd8381383ee08e3f97174ff918b73664ba1b53019e10c76e5cR26-R40 for an example.

Reply:
Ah always forget that I can collect into a result. I'll amend the commit, this looks much better IMO

Reply:
Didn't amend, added a separate commit